### DISCUSSION TITLE
Tuner X Generator

### DISCUSSION THREAD

**Discussion by uwuscutely** [BroncBotz3481/YAGSL-Example#179] [Q&A] [answered]
Hello,

Previously, our team had a full NEO MK4i drive base with a Pigeon 2 on the Rio CAN bus. Now we swapped to full Kraken with a CANivore. What's the difference between using the Tuner X Generator vs the YAGSL one? Will the code base be the same? Are the files generated compatible with YAGSL?

**Comment by thenetworkgrinch** [comment]
They are fundamentally different. Please read CTRE documentation to learn how to use it.

**Reply by uwuscutely** [reply]
Would the Tuner X generator be capable of generating a valid config for YAGSL? Or does it mean I'm switching code bases too?

**Reply by thenetworkgrinch** [reply]
Completely different codebase

**Reply by uwuscutely** [reply]
I see. I saw in docs that if we're using all CTRE compatible devices we should use the Tuner X generator. Is there a specific advantage using that or can I stay with YAGSL if I choose to

**Reply by thenetworkgrinch** [reply, marked as answer]
You can using YAGSL if you choose to, however the Tuner X Swerve Generator is optimjzed. fro all CTRE Products and performs better than YAGSL could bc it eeks out every little optimization they made and the configuration tool is much easier

**Reply by uwuscutely** [reply]
Got it thank you!
